cms/src/db/repositories/baseRepository.ts

```diff
@@ -19,6 +19,44 @@ export class BaseRepository {
         return db.docs.where("parentId").equals(parentId);
     }
 
+    whereParentIds(parentIds: Uuid[]) {
+        return db.docs.where("parentId").anyOf(parentIds);
+    }
+
+    whereLanguageIds(languageIds: Uuid[]) {
+        return db.docs.where("language").anyOf(languageIds);
+    }
+
+    /**
+     * Return a list of documents and change documents of specified DocType that are NOT members of the given groupIds
+     */
+    whereNotMemberOf(groupIds: Array<Uuid>, docType: DocType) {
+        // Query groups and group changeDocs
+        if (docType === DocType.Group) {
+            return db.docs.filter(
+                (group) =>
+                    (group.type === DocType.Group ||
+                        (group.type === DocType.Change && group.docType === DocType.Group)) &&
+                    group.acl! &&
+                    !group.acl.some(
+                        (acl) =>
+                            acl.type === DocType.Group &&
+                            groupIds.includes(acl.groupId) &&
+                            acl.permission.some((p) => p === "view"),
+                    ),
+            );
+        }
+
+        // Query other documents
+        return db.docs.filter(
+            (doc) =>
+                doc.memberOf !== undefined &&
+                (doc.type === docType ||
+                    (doc.type === DocType.Change && doc.docType === docType)) &&
+                !doc.memberOf.some((groupId) => groupIds.includes(groupId)),
```

Comment:
Suggestion:  These filter clauses are a bit hard to read. It might help to split them up in local variables with clearer names.

Reply:
Rewritten to be a bit more readable.